src/directives/badge.js

```diff
@@ -3,36 +3,28 @@ const checkNumber = function (num) {
   return num > 0
 }
 
-const dataBadgeSetter = function (hide, value) {
-  if (hide) {
-    this.el.removeAttribute('data-badge')
-  } else if (!this.isNumber || checkNumber(value)) {
-    this.el.setAttribute('data-badge', value)
+const updateBadgeValue = function (el, binding) {
+  if (!binding.modifiers.number || checkNumber(binding.value)) {
```

Comment:
Where did the hide param go?

I think we should keep this on a separate file as I plan to keep supporting vue 1 too.